pkg/scheduler/apis/config/v1alpha1/defaults.go

```diff
@@ -55,22 +58,37 @@ func SetDefaults_KubeSchedulerConfiguration(obj *kubeschedulerconfigv1alpha1.Kub
 		}
 	}
 
-	if host, port, err := net.SplitHostPort(obj.HealthzBindAddress); err == nil {
+	defaultBindAddress := net.JoinHostPort("0.0.0.0", strconv.Itoa(ports.InsecureSchedulerPort))
+	if obj.HealthzBindAddress == nil {
+		obj.HealthzBindAddress = &defaultBindAddress
+	}
+	if host, port, err := net.SplitHostPort(*obj.HealthzBindAddress); err == nil {
 		if len(host) == 0 {
 			host = "0.0.0.0"
 		}
-		obj.HealthzBindAddress = net.JoinHostPort(host, port)
-	} else {
-		obj.HealthzBindAddress = net.JoinHostPort("0.0.0.0", strconv.Itoa(ports.InsecureSchedulerPort))
+		hostPort := net.JoinHostPort(host, port)
+		obj.HealthzBindAddress = &hostPort
 	}
 
-	if host, port, err := net.SplitHostPort(obj.MetricsBindAddress); err == nil {
+	if obj.MetricsBindAddress == nil {
+		obj.MetricsBindAddress = &defaultBindAddress
+	}
+	if host, port, err := net.SplitHostPort(*obj.MetricsBindAddress); err == nil {
 		if len(host) == 0 {
 			host = "0.0.0.0"
 		}
-		obj.MetricsBindAddress = net.JoinHostPort(host, port)
-	} else {
-		obj.MetricsBindAddress = net.JoinHostPort("0.0.0.0", strconv.Itoa(ports.InsecureSchedulerPort))
```

Comment:
this case got dropped as well, an existing value without a port should get the default port appended